tests/unit/test_column_noise.py

```diff
@@ -126,9 +124,146 @@ def test_miswrite_zipcodes():
+    data = data.astype(int).astype(str)
+    data[data == "-1"] = ""
+    noised_data = _validate_seed_and_noise_data(
+        noise_type=NOISE_TYPES.AGE_MISWRITING, column=data, config=config
+    )
+
+    # Check for expected noise level
+    not_missing_idx = data.index[data != ""]
+    expected_noise = config["row_noise_level"]
+    # todo: Update when generate_incorrect_selection uses exclusive resampling
+    actual_noise = (noised_data[not_missing_idx] != data[not_missing_idx]).mean()
+    # NOTE: we increase the relative tolerance a bit here because the expected
+    # noise calculated above does not account for the fact that if a perturbed
+    # age ends up being the same as the original age, then 1 is subtracted.
+    assert np.isclose(expected_noise, actual_noise, rtol=0.03)
+
+    # Check that missing data remains missing
+    original_missing_idx = data.index[data == ""]
+    noised_missing_idx = noised_data.index[noised_data == ""]
+    pd.testing.assert_index_equal(original_missing_idx, noised_missing_idx)
+
+    # Check that there are no negative ages generated
+    assert noised_data[not_missing_idx].astype(int).min() >= 0
+
+
+def test_miswrite_ages_uniform_probabilities(default_configuration):
+    """Test that an empty list passed in for probabilities results in uniform"""
+    num_rows = 1_000_000
+    age = 25
+    perturbations = [-2, -1, 1]
+
+    config = default_configuration["decennial_census"]["age"]["age_miswriting"]
+    config.update(
+        {
+            "row_noise_level": 1,
+            "possible_perturbations": perturbations,
+        },
+    )
+
+    data = pd.Series([str(age)] * num_rows)
+    noised_data = NOISE_TYPES.AGE_MISWRITING(data, config, RANDOMNESS0, "test")
+    expected_noise = 1 / len(perturbations)
+    for perturbed_age in [str(x + 25) for x in perturbations]:
+        actual_noise = (noised_data == perturbed_age).mean()
+        assert np.isclose(actual_noise, expected_noise, rtol=0.01)
+
+
+def test_miswrite_ages_provided_probabilities(default_configuration):
+    """Test that provided age perturation probabilites are handled"""
+    num_rows = 1_000_000
+    age = 25
+    perturbations = [-1, 1]
+    probabilities = [0.1, 0.9]
+
+    config = default_configuration["decennial_census"]["age"]["age_miswriting"]
+    config.update(
+        {
+            "row_noise_level": 1,
+            "possible_perturbations": perturbations,
+            "possible_perturbation_levels": probabilities,
+        },
+    )
+
+    data = pd.Series([str(age)] * num_rows)
+    noised_data = NOISE_TYPES.AGE_MISWRITING(data, config, RANDOMNESS0, "test")
+    for i in range(len(perturbations)):
+        expected_noise = probabilities[i]
+        perturbed_age = str(age + perturbations[i])
+        actual_noise = (noised_data == perturbed_age).mean()
+        assert np.isclose(actual_noise, expected_noise, rtol=0.01)
+
+
+def test_miswrite_ages_handles_perturbation_to_same_age(default_configuration):
```

Comment:
Should add an equivalent test that negative values are flipped to positive. Maybe everyone has age 2 and gets their age reduced by 5 so everyone should now have an age of 3?